persistence/data_models.go

```diff
@@ -51,51 +51,51 @@ type (
 		NotExists bool
 	}
 
-	GetWorkerTasksRequest struct {
+	GetImmediateTasksRequest struct {
 		ShardId                int32
 		StartSequenceInclusive int64
 		PageSize               int32
 	}
 
-	GetWorkerTasksResponse struct {
-		Tasks []WorkerTask
+	GetImmediateTasksResponse struct {
+		Tasks []ImmediateTask
 		// MinSequenceInclusive is the sequence of first task in the order
 		MinSequenceInclusive int64
 		// MinSequenceInclusive is the sequence of last task in the order
 		MaxSequenceInclusive int64
 	}
 
-	DeleteWorkerTasksRequest struct {
+	DeleteImmediateTasksRequest struct {
 		ShardId int32
 
 		MinTaskSequenceInclusive int64
 		MaxTaskSequenceInclusive int64
 	}
 
-	BackoffWorkerTaskRequest struct {
+	BackoffImmediateTaskRequest struct {
 		LastFailureStatus    int32
 		LastFailureDetails   string
 		Prep                 PrepareStateExecutionResponse
-		Task                 WorkerTask
+		Task                 ImmediateTask
 		FireTimestampSeconds int64
 	}
 
-	ConvertTimerTaskToWorkerTaskRequest struct {
+	ConvertTimerTaskToImmediateTaskRequest struct {
 		Task TimerTask
 	}
 
-	WorkerTask struct {
+	ImmediateTask struct {
 		ShardId int32
 		// TaskSequence represents the increasing order in the queue of the shard
 		// It should be empty when inserting, because the persistence/database will
 		// generate the value automatically
 		TaskSequence *int64
 
-		TaskType WorkerTaskType
+		TaskType ImmediateTaskType
 
-		ProcessExecutionId uuid.UUID
```

Comment:
I don't think this needs to be renamed. 
It's to tell which processExecutionId the task is for. E.g. for LocalQueueMessageReceived task, we also need a process exe Id.

Reply:
For LocalQueueMessageReceived, I think we just need to store the dedupId, then we can get the sent message from the local_queue table by the dedupId.

Reply:
ah right, we will need both the PK in local_queue table is <proc_exe_id, queueName, dedup_id> based on our doc :P

Reply:
keep process_execution_id in the woker table acutally can save one query to get the process_execution_id from local_queue table. Yeah it's worth adding it back.